sim/compare_snps.py

```diff
@@ -18,25 +19,46 @@ def masked_positions(mask_filepath):
     #    why is that?
     masked_pos = []
     for i, row in mask.iterrows():
-        masked_pos.extend(list(range(row['START'], row['END']+1)))
+        masked_pos.extend(list(range(row['START'], row['END']+1)))# This is based off indexing in richards rpt.regions file - inconsisten with bedtools
 
     return masked_pos
 
-def analyse(simulated_snps, pipeline_snps, mask_filepath):
+def analyse(results_path, sample_name, mask_filepath):
     """ Compare simulated SNPs data from simuG against btb-seq's snpTable.tab
         If adjust == True: applies the mask to simulated SNPs and pipeline SNPs
         Returns a dictionary of performance stats
     """
 
+    pipeline_directory = glob.glob(results_path + 'btb-seq-results/Results_simulated-reads_*')[0] + '/'
+    simulated_snp_path = results_path + f'simulated-genome/{sample_name}.simulated.refseq2simseq.map.txt'
+    pipeline_snp_path = pipeline_directory + f'snpTables/{sample_name}_snps.tab'
+    if not os.path.exists(pipeline_snp_path):
+        pipeline_snp_path = pipeline_directory + f'snpTables/{sample_name}.tab'
+    if not os.path.exists(pipeline_snp_path):
+        raise Exception("Cant Find the pipeline's snps table!!")
+    pipeline_genome_path = pipeline_directory + f'consensus/{sample_name}_consensus.fas'
+    if not os.path.exists(pipeline_genome_path):
+        pipeline_genome_path = pipeline_directory + f'consensus/{sample_name}.fas'
+    if not os.path.exists(pipeline_snp_path):
+        raise Exception("Cant Find the pipeline's consensus file!!")
     # Load
-    simulated = pd.read_csv(simulated_snps, delimiter='\t')
-    pipeline = pd.read_csv(pipeline_snps, delimiter='\t')
+    simulated_snps = pd.read_csv(simulated_snp_path,  delimiter='\t')
+    pipeline_snps = pd.read_csv(pipeline_snp_path, delimiter='\t')
+    pipeline_genome = load_consensus(pipeline_genome_path)
+
+    # Trim adapter sequences off pipeline genome
+    pipeline_genome = pipeline_genome[29:-31]
 
     # Extract SNP positions
-    simulated_pos = set(simulated['ref_start'].values)
-    pipeline_pos = set(pipeline['POS'].values)
+    simulated_pos = set(simulated_snps['ref_start'].values)
+    pipeline_pos = set(pipeline_snps['POS'].values)
+    
+    # Extract mask positions    
     masked_pos = set(masked_positions(mask_filepath))
 
+    # Extract N's positions in pipeline genome
+    n_pos = set([i+30 for i in range(len(pipeline_genome)) if pipeline_genome[i] == 'N'])
```

Comment:
another hardcoded value, based off the previous

Reply:
This +30 is to acheive a consistent indexing with the positions in the map.txt and .tab files, which start at 30 (I guess due to the adapter sequence).

Reply:
I see, perhaps just put a comment above to saying that, will make it easier to cleanup later